submissions/zhenyakornilov/memory-pair-game/js/index.js

```diff
@@ -0,0 +1,107 @@
+import cards from "./cards.js";
+
+const cardsMenu = document.getElementById("game-cards");
+const totalTriesCounter = document.getElementById("total-flips");
+const states = {
+  cardsMatches: 0,
+  firstFlippedCard: null,
+  secondFlippedCard: null,
+  currentCardPair: [],
+  totalTries: 0,
+};
+const allGameCards = shuffleCards([...cards, ...cards]);
+renderCards(allGameCards, cardsMenu);
+
+function shuffleCards(cardsArr) {
+  for (let i = cardsArr.length - 1; i > 0; i--) {
+    let j = Math.floor(Math.random() * (i + 1));
+    let swap = cardsArr[i];
+    cardsArr[i] = cardsArr[j];
+    cardsArr[j] = swap;
+  }
+
+  return cardsArr;
+}
+
+function renderCards(cardsObj, cardsList) {
+  let cardsHTML = cardsObj
+    .map(({ name, imageSrc }) => {
+      return `
+      <li class="game-card">
+        <div class="card-inner">
+          <div class="card-front">
+            <img class="card-img-front" src="images/treble_clef.webp" alt="image of treble clef">
+          </div>
+          <div class="card-back">
+            <img class="card-img-back" src="${imageSrc}" alt="image of ${name}">
+            <p class="card-desc">${name}</p>
+          </div>
+        </div>
+      </li>
+      `;
+    })
+    .join("");
+
+  cardsList.innerHTML = cardsHTML;
+}
+
+function sleep(milliseconds) {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds));
```

Comment:
why do we need async operation here?

Reply:
I wanted to create some kind of custom delay function instead of repeating **setTimeout** all the time. This is solution I came up with and thats why I use async - await later. I haven't figured out how to do that in another way yet (except using setTimeout in every place where I currently used my **sleep()** function.